src/mofdscribe/featurizers/utils/substructures.py

```diff
@@ -35,6 +35,39 @@ def filter_element(
         return Structure.from_sites(keep_sites)
     else:  # input is molecule or IMolecule
         return Molecule.from_sites(keep_sites)
+    
+
+def filter_element_for_ph(
+    structure: Union[Structure, IStructure, Molecule, IMolecule], elements: str
+) -> Structure:
+    """Filter a structure by element.
+
+    Args:
+        structure (Union[Structure, IStructure, Molecule, IMolecule]): input structure
+        elements (str): element to filter
+
+    Returns:
+        filtered_structure (Structure): filtered structure
+    """
+    elements_ = []
+    elements_group = (elements,)
+    for atom_type in elements_group:
```

Comment:
i'm not sure I understand this change. You are certainly right that the old code behaves in an unexpected way for elements with more than two symbols in the letter. But wouldn't then the clearer fix be something else? For example, always split by `-` first and then handle other things